rplugin/python3/deoplete/deoplete.py

```diff
@@ -53,6 +55,13 @@ def __init__(self, vim):
         self.__profile_start = 0
 
     def completion_begin(self, context):
+        if not logger.init:
+            log_file = None
+            if self.__vim.vars['deoplete#enable_debug']:
+                self.debug_enabled = True
+                log_file = self.__vim.vars['deoplete#debug_log']
+            logger.setup(self.__vim, log_file)
```

Comment:
The reason I did this is because of `self.__vim.vars['deoplete#enable_debug']` availability.  I also wanted it to be checked only once.  When `Deoplete.__init__()` is called, `deoplete#enable_debug` is not available until `call _deoplete()` occurs in VimL.  The other issue is that an instance of `vim` needs to be passed to `setup()` so that it's able to print errors in Vim.  I think it looks strange because of the conditions that need to be met for how I wrote it.

Maybe it's better to have the logging filter check `deoplete#enable_debug` lazily when the first message is emitted?  This way, `setup()` can be called in `Deoplete.__init__()` without initial checks.  What do you think?